Yoast/wordpress-seo
# Send SEO title and meta description without replace vars to analysis.

## Summary

This PR can be summarized in the following changelog entry:

* [not-user-facing] The SEO title and meta description without replace vars, the page title and the excerpt are now send to the SEO and Readability analysis.

## Relevant technical choices:

* I decided to add the post/page title and excerpt for now, since they were the most straightforward to implement.
   * They were already available in the edit store.
   * Retrieving the categories and primary category of the page is more difficult, since they are not yet available in the YoastSEO redux store given to the `collectAnalysisData` method. 
      * I will make an issue to implement this.

## Test instructions


This PR can be tested by following these steps:
* Start your local `wordpress-seo` development environment.
* Make sure that the `define( 'YOAST_SEO_DEBUG_ANALYSIS_WORKER', 'TRACE' );` flag in your `wp-config.php` file is set to `'TRACE'` to be able to see what is being sent to the analysis web worker. 
* Open a post.

### Test if the SEO title, without replacement variables, is sent to the analysis.
* Add some replacement variables to the SEO title of the post.
   * The SEO title editor can be found by clicking on 'edit snippet' inside the snippet editor in the YoastSEO metabox.
   * Try out different replacement variables, like `Site Title` and `Date`.
* Add some normal text to the SEO title as well, intermixed with the replace vars.
* Open up the browser's development console.
* You should see one or multiple analysis requests going to the analysis web worker.
   * They have the form: `AnalysisWebWorker incoming: analyze 7 {paper: Paper}`.
* Open the paper that is sent to the analysis in the aforementioned analysis request. (by clicking on the `{paper: Paper}` item in the console)
   * It should have an item under `_attributes` called `titleWithoutReplacements`.
   * This item should contain the text of the SEO title, without the replacements from the replace vars.

### Test if the meta description title, without replacement variables, is sent to the analysis.
* Add some replacement variables to the meta description of the post.
   * The meta description editor can be found by clicking on 'edit snippet' inside the snippet editor in the YoastSEO metabox.
   * Try out different replacement variables, like `Site Title` and `Date`.
* Add some normal text to the meta description as well, intermixed with the replace vars.
* Open up the browser's development console.
* You should see one or multiple analysis requests going to the analysis web worker.
   * They have the form: `AnalysisWebWorker incoming: analyze 7 {paper: Paper}`.
* Open the paper that is sent to the analysis in the aforementioned analysis request. (by clicking on the `{paper: Paper}` item in the console)
   * It should have an item under `_attributes` called `descriptionWithoutReplacements`.
   * This item should contain the text of the meta description, without the replacements from the replace vars.

### Test if the page title is sent to the analysis.
* Add a title to the post, if the post does not have one yet.
  * In this case, the title is the actual post title, not the SEO title!
* The paper that is sent to the analysis should contain an attribute called `pageTitle`.
* The attribute's content should be the same as the post's title. 

### Test if the excerpt is sent to the analysis.
* Add an excerpt to the post, if the post does not have one yet.
  * The excerpt editor can be found in the sidebar of the Gutenberg editor.
     * On the 'Document' tab, inside of the 'Excerpt' collapsible.
* The paper that is sent to the analysis should contain an attribute called `excerpt`.
* The attribute's content should be the same as the excerpt. 

### Test if not internal linking suggestions are computed when the post is empty.
See https://github.com/Yoast/wordpress-seo-premium/issues/2417.
* Open a new post.
* Wait 10 seconds.
* You should see no internal linking suggestions (since the post has no content yet).
* Add some content to the post.
* Wait a few seconds.
* You should see internal linking suggestions.

## UI changes

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2401
Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2417
